ambry-frontend/src/main/java/com/github/ambry/frontend/s3/S3BatchDeleteHandler.java

```diff
@@ -0,0 +1,179 @@
+      }
+
+      S3MessagePayload.DeleteResult response = new S3MessagePayload.DeleteResult();
+      ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+      ReadableStreamChannel readableStreamChannel =
+          new ByteBufferReadableStreamChannel(ByteBuffer.wrap(outputStream.toByteArray()));
+      S3MessagePayload.S3BatchDeleteObjects deleteRequest = deserializeRequest(channel);
+      ConcurrentLinkedQueue<S3MessagePayload.S3ErrorObject> errors = new ConcurrentLinkedQueue<>();
+      ConcurrentLinkedQueue<S3MessagePayload.S3DeletedObject> deleted = new ConcurrentLinkedQueue<>();
+      RequestPath requestPath = (RequestPath) restRequest.getArgs().get(InternalKeys.REQUEST_PATH);
+      if (deleteRequest.getObjects().size() > MAX_BATCH_DELETE_SIZE) {
+        String batchSizeErrorMessage = "Exceeded Maximum Batch Size of ";
+        LOGGER.error(batchSizeErrorMessage, MAX_BATCH_DELETE_SIZE);
+        throw new RestServiceException(batchSizeErrorMessage, RestServiceErrorCode.BadRequest);
+      }
+
+      List<CompletableFuture<Void>> deleteFutures = new ArrayList<>();
+      for (S3MessagePayload.S3BatchDeleteKey object : deleteRequest.getObjects()) {
+        String singleDeletePath = requestPath.getPathAfterPrefixes() + "/" + object.getKey();
+        RequestPath newRequestPath =
+            RequestPath.parse(singleDeletePath, restRequest.getArgs(), new ArrayList<>(), requestPath.getClusterName());
+        WrappedRestRequest singleDeleteRequest = new WrappedRestRequest(restRequest);
+        singleDeleteRequest.setArg(InternalKeys.REQUEST_PATH, newRequestPath);
+        NoOpResponseChannel noOpResponseChannel = new NoOpResponseChannel();
+        CompletableFuture<Void> future = new CompletableFuture<>();
+
+        // Handle the delete operation using the deleteBlobHandler
+        deleteBlobHandler.handle(singleDeleteRequest, noOpResponseChannel, (result, exception) -> {
+          // Call our custom onDeleteCompletion to track success/failure
+          if (exception == null) {
+            deleted.add(new S3MessagePayload.S3DeletedObject(object.getKey()));
+          }
+          else if (exception instanceof RestServiceException) {
+            RestServiceException restServiceException = (RestServiceException) exception;
+            errors.add((new S3MessagePayload.S3ErrorObject(object.getKey(), restServiceException.getErrorCode().toString())));
+          }
+          else {
+            errors.add((new S3MessagePayload.S3ErrorObject(object.getKey(), RestServiceErrorCode.InternalServerError.toString())));
+          }
+          future.complete(null);
+        });
+        deleteFutures.add(future);
+      }
+
+      CompletableFuture.allOf(deleteFutures.toArray(new CompletableFuture[0]))
+          .whenComplete((result, exception) -> {
+            try {
+              // Add XML declaration at the top
+              String xmlDeclaration = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n";
```

Comment:
Let' use xmlMapper's configuration for this task
```
xmlMapper.configure(ToXmlGenerator.Feature.WRITE_XML_DECLARATION, true);
```
when creating this xmlMapper